### ISSUE TITLE
zipPartitions issue for reads data on cluster

### ISSUE THREAD

**Issue by erictu** [bigdatagenomics/mango#113]
This is when loading in reads data (issuing reads http request)

```
java.lang.IllegalArgumentException: Can't zip RDDs with unequal numbers of partitions
	at org.apache.spark.rdd.ZippedPartitionsBaseRDD.getPartitions(ZippedPartitionsRDD.scala:57)
	at org.apache.spark.rdd.RDD$$anonfun$partitions$2.apply(RDD.scala:239)
	at org.apache.spark.rdd.RDD$$anonfun$partitions$2.apply(RDD.scala:237)
	at scala.Option.getOrElse(Option.scala:120)
	at org.apache.spark.rdd.RDD.partitions(RDD.scala:237)
	at org.apache.spark.rdd.MapPartitionsRDD.getPartitions(MapPartitionsRDD.scala:35)
	at org.apache.spark.rdd.RDD$$anonfun$partitions$2.apply(RDD.scala:239)
	at org.apache.spark.rdd.RDD$$anonfun$partitions$2.apply(RDD.scala:237)
	at scala.Option.getOrElse(Option.scala:120)
	at org.apache.spark.rdd.RDD.partitions(RDD.scala:237)
	at org.apache.spark.rdd.MapPartitionsRDD.getPartitions(MapPartitionsRDD.scala:35)
	at org.apache.spark.rdd.RDD$$anonfun$partitions$2.apply(RDD.scala:239)
	at org.apache.spark.rdd.RDD$$anonfun$partitions$2.apply(RDD.scala:237)
	at scala.Option.getOrElse(Option.scala:120)
	at org.apache.spark.rdd.RDD.partitions(RDD.scala:237)
	at org.apache.spark.rdd.MapPartitionsRDD.getPartitions(MapPartitionsRDD.scala:35)
	at org.apache.spark.rdd.RDD$$anonfun$partitions$2.apply(RDD.scala:239)
	at org.apache.spark.rdd.RDD$$anonfun$partitions$2.apply(RDD.scala:237)
	at scala.Option.getOrElse(Option.scala:120)
	at org.apache.spark.rdd.RDD.partitions(RDD.scala:237)
	at org.apache.spark.rdd.MapPartitionsRDD.getPartitions(MapPartitionsRDD.scala:35)
	at org.apache.spark.rdd.RDD$$anonfun$partitions$2.apply(RDD.scala:239)
	at org.apache.spark.rdd.RDD$$anonfun$partitions$2.apply(RDD.scala:237)
	at scala.Option.getOrElse(Option.scala:120)
	at org.apache.spark.rdd.RDD.partitions(RDD.scala:237)
	at org.apache.spark.rdd.MapPartitionsRDD.getPartitions(MapPartitionsRDD.scala:35)
	at org.apache.spark.rdd.RDD$$anonfun$partitions$2.apply(RDD.scala:239)
	at org.apache.spark.rdd.RDD$$anonfun$partitions$2.apply(RDD.scala:237)
	at scala.Option.getOrElse(Option.scala:120)
	at org.apache.spark.rdd.RDD.partitions(RDD.scala:237)
	at org.apache.spark.SparkContext.runJob(SparkContext.scala:1921)
	at org.apache.spark.rdd.RDD$$anonfun$collect$1.apply(RDD.scala:905)
	at org.apache.spark.rdd.RDDOperationScope$.withScope(RDDOperationScope.scala:147)
	at org.apache.spark.rdd.RDDOperationScope$.withScope(RDDOperationScope.scala:108)
	at org.apache.spark.rdd.RDD.withScope(RDD.scala:306)
	at org.apache.spark.rdd.RDD.collect(RDD.scala:904)
	at org.bdgenomics.mango.layout.AlignmentRecordLayout$.apply(AlignmentRecordLayout.scala:54)
	at org.bdgenomics.mango.cli.VizServlet$$anonfun$4$$anonfun$apply$2.apply(VizReads.scala:268)
	at scala.Option.fold(Option.scala:157)
	at org.apache.spark.rdd.Timer.time(Timer.scala:48)
	at org.bdgenomics.mango.cli.VizServlet$$anonfun$4.apply(VizReads.scala:250)
	at org.scalatra.ScalatraKernel$class.org$scalatra$ScalatraKernel$$liftAction(ScalatraKernel.scala:216)
	at org.scalatra.ScalatraKernel$$anonfun$invoke$1.apply(ScalatraKernel.scala:211)
	at org.scalatra.ScalatraKernel$$anonfun$invoke$1.apply(ScalatraKernel.scala:211)
	at org.scalatra.ScalatraKernel$class.withRouteMultiParams(ScalatraKernel.scala:274)
	at org.scalatra.ScalatraServlet.withRouteMultiParams(ScalatraServlet.scala:24)
	at org.scalatra.ScalatraKernel$class.invoke(ScalatraKernel.scala:210)
	at org.scalatra.ScalatraServlet.invoke(ScalatraServlet.scala:24)
	at org.scalatra.ScalatraKernel$$anonfun$runRoutes$1$$anonfun$apply$2.apply(ScalatraKernel.scala:196)
	at org.scalatra.ScalatraKernel$$anonfun$runRoutes$1$$anonfun$apply$2.apply(ScalatraKernel.scala:195)
	at scala.Option.flatMap(Option.scala:170)
	at org.scalatra.ScalatraKernel$$anonfun$runRoutes$1.apply(ScalatraKernel.scala:195)
	at org.scalatra.ScalatraKernel$$anonfun$runRoutes$1.apply(ScalatraKernel.scala:194)
	at scala.collection.immutable.Stream.flatMap(Stream.scala:446)
	at org.scalatra.ScalatraKernel$class.runRoutes(ScalatraKernel.scala:194)
	at org.scalatra.ScalatraServlet.runRoutes(ScalatraServlet.scala:24)
	at org.scalatra.ScalatraKernel$class.executeRoutes(ScalatraKernel.scala:165)
	at org.scalatra.ScalatraServlet.executeRoutes(ScalatraServlet.scala:24)
	at org.scalatra.ScalatraKernel$$anonfun$handle$1$$anonfun$apply$mcV$sp$1.apply$mcV$sp(ScalatraKernel.scala:139)
	at org.scalatra.ScalatraKernel$$anonfun$handle$1$$anonfun$apply$mcV$sp$1.apply(ScalatraKernel.scala:137)
	at org.scalatra.ScalatraKernel$$anonfun$handle$1$$anonfun$apply$mcV$sp$1.apply(ScalatraKernel.scala:137)
	at scala.util.DynamicVariable.withValue(DynamicVariable.scala:57)
	at org.scalatra.ScalatraKernel$$anonfun$handle$1.apply$mcV$sp(ScalatraKernel.scala:137)
	at org.scalatra.ScalatraKernel$$anonfun$handle$1.apply(ScalatraKernel.scala:137)
	at org.scalatra.ScalatraKernel$$anonfun$handle$1.apply(ScalatraKernel.scala:137)
	at scala.util.DynamicVariable.withValue(DynamicVariable.scala:57)
	at org.scalatra.ScalatraKernel$class.handle(ScalatraKernel.scala:136)
	at org.scalatra.ScalatraServlet.handle(ScalatraServlet.scala:24)
	at org.scalatra.ScalatraServlet.service(ScalatraServlet.scala:31)
	at javax.servlet.http.HttpServlet.service(HttpServlet.java:848)
	at org.eclipse.jetty.servlet.ServletHolder.handle(ServletHolder.java:565)
	at org.eclipse.jetty.servlet.ServletHandler.doHandle(ServletHandler.java:479)
	at org.eclipse.jetty.server.handler.ScopedHandler.handle(ScopedHandler.java:119)
	at org.eclipse.jetty.security.SecurityHandler.handle(SecurityHandler.java:521)
	at org.eclipse.jetty.server.session.SessionHandler.doHandle(SessionHandler.java:227)
	at org.eclipse.jetty.server.handler.ContextHandler.doHandle(ContextHandler.java:1031)
	at org.eclipse.jetty.servlet.ServletHandler.doScope(ServletHandler.java:406)
	at org.eclipse.jetty.server.session.SessionHandler.doScope(SessionHandler.java:186)
	at org.eclipse.jetty.server.handler.ContextHandler.doScope(ContextHandler.java:965)
	at org.eclipse.jetty.server.handler.ScopedHandler.handle(ScopedHandler.java:117)
	at org.eclipse.jetty.server.handler.ContextHandlerCollection.handle(ContextHandlerCollection.java:250)
	at org.eclipse.jetty.server.handler.HandlerWrapper.handle(HandlerWrapper.java:111)
	at org.eclipse.jetty.server.Server.handle(Server.java:349)
	at org.eclipse.jetty.server.AbstractHttpConnection.handleRequest(AbstractHttpConnection.java:449)
	at org.eclipse.jetty.server.AbstractHttpConnection$RequestHandler.headerComplete(AbstractHttpConnection.java:910)
	at org.eclipse.jetty.http.HttpParser.parseNext(HttpParser.java:634)
	at org.eclipse.jetty.http.HttpParser.parseAvailable(HttpParser.java:230)
	at org.eclipse.jetty.server.AsyncHttpConnection.handle(AsyncHttpConnection.java:76)
	at org.eclipse.jetty.io.nio.SelectChannelEndPoint.handle(SelectChannelEndPoint.java:609)
	at org.eclipse.jetty.io.nio.SelectChannelEndPoint$1.run(SelectChannelEndPoint.java:45)
	at org.eclipse.jetty.util.thread.QueuedThreadPool.runJob(QueuedThreadPool.java:599)
	at org.eclipse.jetty.util.thread.QueuedThreadPool$3.run(QueuedThreadPool.java:534)
	at java.lang.Thread.run(Thread.java:745)

```

**Comment by erictu** [comment]
Error is with regards to checking for files. Currently for reads data (and for loading bams) we check the local file system, when we should be checking the distributed file system.

**Comment by erictu** [comment]
resolved by #119
